build.gradle.kts

```diff
@@ -276,9 +290,40 @@ tasks.shadowJar {
 			"TweakClass" to "org.spongepowered.asm.launch.MixinTweaker",
 		)
 	}
+	
+	doLast {
+		// re-zip the jar with highest compression level
+		val jarFile = shadowJar.archiveFile.get().asFile
+		val fileContents = mutableMapOf<String, ByteArray>()
+		ZipInputStream(jarFile.inputStream()).use { zis ->
+			var entry: ZipEntry? = zis.nextEntry
+			while (entry != null) {
+				if (!entry.isDirectory) {
+					val bytes = zis.readAllBytes()
+					fileContents[entry.name] = bytes
+				}
+				entry = zis.nextEntry
+			}
+		}
+
+		ZipOutputStream(Files.newOutputStream(jarFile.toPath())).use { zos ->
+			zos.setLevel(Deflater.BEST_COMPRESSION)
+			fileContents.forEach { (name, bytes) ->
+				var newBytes = bytes
+				if(name.endsWith(".json") || name.endsWith(".mcmeta") || name.endsWith("mcmod.info")) {
+					newBytes = JsonOutput.toJson(JsonSlurper().parse(bytes)).toByteArray()
+				}
+				zos.putNextEntry(ZipEntry(name))
+				zos.write(newBytes)
+				zos.closeEntry()
+			}
+			zos.finish()
+			zos.close()
+		}
+	}
 }
 
-tasks.build {
+tasks.assemble {
```

Comment:
I'd like to understand what the impact of this change is before merging it.

Reply:
more of a personal change I made, assemble just makes the jar, build makes it and tests it (and there are no tests here, so it's basically the same)
build depends on assemble so it functionally should have no difference if you are running build, but if you run just assemble then it works correctly